connectors/spark/src/test/java/io/unitycatalog/spark/auth/AwsCredentialRenewalTest.java

```diff
@@ -0,0 +1,362 @@
+import io.unitycatalog.server.service.credential.aws.CredentialsGenerator;
+import io.unitycatalog.spark.CredentialTestFileSystem;
+import io.unitycatalog.spark.UCHadoopConf;
+import io.unitycatalog.spark.UCSingleCatalog;
+import io.unitycatalog.spark.utils.Clock;
+import java.io.File;
+import java.net.URI;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.api.java.function.MapPartitionsFunction;
+import org.apache.spark.sql.Encoders;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.util.SerializableConfiguration;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.sparkproject.guava.collect.ImmutableList;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+public class AwsCredentialRenewalTest extends BaseCRUDTest {
+
+  private static final String CLOCK_NAME = AwsCredentialRenewalTest.class.getSimpleName();
+  private static final String CREDENTIALS_GENERATOR = TimeBasedCredentialsGenerator.class.getName();
+
+  private static final String CATALOG_NAME = "CredRenewalCatalog";
+  private static final String SCHEMA_NAME = "Default";
+  private static final String TABLE_NAME = String.format("%s.%s.demo", CATALOG_NAME, SCHEMA_NAME);
+  private static final String BUCKET_NAME = "test-bucket";
+  private static final String S3_SCHEME = "s3:";
+  private static final long DEFAULT_INTERVAL_MILLIS = 30_000L;
+
+  @TempDir private File dataDir;
+  private SparkSession session;
+  private SdkSchemaOperations schemaOperations;
+
+  @Override
+  protected CatalogOperations createCatalogOperations(ServerConfig config) {
+    return new SdkCatalogOperations(createApiClient(config));
+  }
+
+  @Override
+  protected void setUpProperties() {
+    super.setUpProperties();
+    serverProperties.put("s3.bucketPath.0", "s3://" + BUCKET_NAME);
+    serverProperties.put("s3.accessKey.0", "accessKey0");
+    serverProperties.put("s3.secretKey.0", "secretKey0");
+    serverProperties.put("s3.sessionToken.0", "sessionToken0");
+    serverProperties.put("s3.credentialsGenerator.0", CREDENTIALS_GENERATOR);
```

Comment:
can you leave a comment here explaining what this will do?

Reply:
Done, thanks for sharing !